### PR TITLE
Added support to change timezone and migrated from moment to luxon (#85zrhudt7)

### LINKED ISSUE #96: Add support to change timezone and migrate from moment to luxon
## What is the motivation for adding/enhancing this feature?

You could take reference from existing PR. Use the latest Luxon library
https://github.com/hotwax/picking/pull/61
## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #96 

### Short Description and Why It's Useful

Added timezone changing support and replaced the use of moment with luxon.

### Screenshots of Visual Changes before/after (If There Are Any)

Picking - HotWax Commerce

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by shashwatbangar** [comment]
Verified workflow. Looks good to me.

**Review by shashwatbangar — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -22,9 +22,8 @@
     "axios-cache-adapter": "^2.7.3",
     "core-js": "^3.6.5",
     "http-status-codes": "^2.1.4",
+    "luxon": "^2.4.0",
```
Comment (adityasharma7) [inline package.json:25]: We could use 3.2 latest

File: src/store/modules/user/actions.ts [hunk 2]
```diff
@@ -83,7 +82,10 @@ const actions: ActionTree<UserState, RootState> = {
   async getProfile ( { commit }) {
     const resp = await UserService.getProfile()
     if (resp.status === 200) {
-      const localTimeZone = moment.tz.guess();
+      if (resp.data.userTimeZone) {
+        Settings.defaultZone = resp.data.userTimeZone;
+      }
+      const localTimeZone = DateTime.local().zoneName;
       if (resp.data.userTimeZone !== localTimeZone) {
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:89]: We could remove this logic

File: src/views/TimezoneModal.vue [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      })
+    },
+    selectSearchBarText(event: any) {
+      event.target.getInputElement().then((element: any) => {
+        element.select();
+      })
+    },
+    async setUserTimeZone() {
+      return this.store.dispatch("user/setUserTimeZone", {
+        "tzId": this.timeZoneId
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:149]:
```suggestion
        "timeZoneId": this.timeZoneId
```

File: src/views/TimezoneModal.vue [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      return alert.present();
+    },
+    escapeRegExp(text: string) {
+      //TODO Handle it in a better way
+      // Currently when the user types special character as it part of Regex expressions it breaks the code
+      // so removed the characters for now
+      return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+    },
+    findTimeZone() { 
+      const regularExp = new RegExp(`${this.escapeRegExp(this.queryString)}`, 'i');
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:125]: We could disallow user to type such characters with keyup/keydown events

File: src/views/TimezoneModal.vue [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+      this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
+        return regularExp.test(timeZone.id) || regularExp.test(timeZone.label);
+      });
+    },
+    async getAvailableTimeZones() {
+      UserService.getAvailableTimeZones().then((resp: any) => {
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:132]: We could use await here

File: src/views/TimezoneModal.vue [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+          // We are filtering valid the timeZones coming with response here
+          this.timeZones = resp.data.filter((timeZone: any) => {
+            return DateTime.local().setZone(timeZone.id).isValid;
+          });
+          this.findTimeZone();
+        }
+      })
+    },
+    selectSearchBarText(event: any) {
+      event.target.getInputElement().then((element: any) => {
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:143]: We could use async-await

**Review by adityasharma7 — APPROVED** [review]

File: src/views/TimezoneModal.vue (hunk 5 above)
Comment (k2maan) [reply src/views/TimezoneModal.vue]: Updated sir

File: src/views/TimezoneModal.vue (hunk 6 above)
Comment (k2maan) [reply src/views/TimezoneModal.vue]: Updated sir

File: src/views/TimezoneModal.vue (hunk 3 above)
Comment (k2maan) [reply src/views/TimezoneModal.vue]: Updated sir

File: src/store/modules/user/actions.ts (hunk 2 above)
Comment (k2maan) [reply src/store/modules/user/actions.ts]: Removed

File: package.json (hunk 1 above)
Comment (k2maan) [reply package.json]: Updated

File: src/views/TimezoneModal.vue (hunk 4 above)
Comment (k2maan) [reply src/views/TimezoneModal.vue]: Done sir

**Pull request closed without merging**